File: frontend/src/plugins/simulators/social_internal/manifest.ts
import { lazy } from 'react'
import { AtSign } from 'lucide-react'
import type { PluginManifest } from '../../types'

export const manifest: PluginManifest = {
  code: 'social_internal',
  name: 'plugins.social_internal.name',
  description: 'plugins.social_internal.description',
  category: 'simulator',
  icon: AtSign,
  defaultColor: 'blue',
  supportedFormats: ['text', 'video', 'image'],
  defaultEnabled: false,
  comingSoon: false,
  sortOrder: 2,
  injectTypes: ['socialnet'],
  playerRoute: '/social',
  PlayerPage: lazy(() => import('./PlayerPage')),
  EventReceiver: lazy(() => import('./EventReceiver')),
  PreviewPopup: lazy(() => import('./PreviewPopup')),
  configSchema: null,
}

File: backend/app/plugins/simulators/social_internal/manifest.py
from app.plugins.base import PluginCategory, PluginManifest

manifest = PluginManifest(
    code="social_internal",
    name="Reseau social",
    name_en="Social Network",
    description="Simulation de fil social interne",
    category=PluginCategory.SIMULATOR,
    icon="Twitter",
    default_color="blue",
    supported_formats=["text", "video", "image"],
    default_enabled=False,
    coming_soon=False,
    sort_order=2,
    inject_types=["socialnet"],
    router_module="app.routers.twitter",
    router_prefix="/api/twitter",
)
